### ISSUE TITLE
[Feature Request] Support memorySwap.swapBehavior in Customize node configuration

### ISSUE THREAD

**Issue by zhiweiv** [Azure/AKS#4352]
We want to set memorySwap.swapBehavior in kubelet configuration to use kubernetes swap feature, currently this one is not in aks whitelist.

References: 
https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration?tabs=linux-node-pools#linux-kubelet-custom-configuration
https://kubernetes.io/blog/2023/08/24/swap-linux-beta/
https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory

**Comment by denniszielke** [comment]
@allyford  Since 1.30 upstream has swap-memory support by default. Would be great to have a custom node config for that.

**Comment by allyford** [comment]
Thanks for creating this issue, I've added this to our backlog. Currently no ETA. Will update this issue if anything changes.

**Comment by mrmiyagietas** [comment]
@allyford  
Would be nice to get notified when there will be some update on this topic :) We also want to use this functionality in the future to improve our setup of self hosted agents for Azure DevOps pipelines and GitHub Action Runners such as Jenkins Agents. There we see much potential for better scalability / performance usage of the nodes.

**Comment by dixneuf19** [comment]
Hi,
Also interested by this feature, which would be really useful for some workfloads running in our AKS clusters.

**Comment by mattmurp** [comment]
@allyford we're interested in this feature, as well. Is there any update as to possible timelines? Thanks.

**Comment by zhiweiv** [comment]
NodeSwap is GA on k8s 1.34, any update on this?

**Comment by allyford** [comment]
We currently have no ETA for adding this field to the Custom Node Configuration feature. However, we are working on an alternative feature which will allow for more flexibility in custom settings. See #4704

**Comment by lucsoft** [comment]
is this feature request now dead?

**Comment by benritchie** [comment]
Agree, this feature request is strongly needed.  its a gap that AKS does support enabling swap on nodes https://docs.azure.cn/en-us/aks/custom-node-configuration?tabs=linux-node-pools ).  - but doesn't allow that swap to actually be used - without allowing swapBehaviour to be set ot limited, enabling swap has little affect.  - and PIS is rather overkill for this, where I think all that's needed is for this config to be added to the whitelist so it can be set through kubelet_config?

**Comment by allyford** [comment]
Thanks for sharing your requirements for this feature. This is in our backlog, but we should have an updated ETA soon. I've reopened the issue and will provide updates here.
